lib/zstd.h

```diff
@@ -520,7 +558,7 @@ ZSTDLIB_API ZSTD_CStream* ZSTD_createCStream_advanced(ZSTD_customMem customMem);
 ZSTDLIB_API size_t ZSTD_initCStream_srcSize(ZSTD_CStream* zcs, int compressionLevel, unsigned long long pledgedSrcSize);   /**< pledgedSrcSize must be correct */
```

Comment:
Here, we have to choose if `pledgedSrcSize==0` really means `0`, or if it means `unknown`.
Both have pros and cons.

Up to now `pledgedSrcSize==0` meant `unknown`, so preserving this behavior would allow existing users to continue working as intended.
But it would make this API unable to generate a `zero-size` in frame header.
To do so, one would need to implement `ZSTD_initCStream_advanced()`.

Changing it to mean `0` could break a few programs, so it needs to be properly documented.
Work-around to mean `unknown` would be to use `ZSTD_initCStream()`.
Existing programs would need to create a branch for this usage.

Due to the risk of breaking existing applications, I would rather lean for the first solution.
To be debated.